akka-tools-persistence/src/main/scala/no/nextgentel/oss/akkatools/persistence/EnhancedPersistentActor.scala

```diff
@@ -151,25 +185,73 @@ abstract class EnhancedPersistentActor[E:ClassTag, Ex <: Exception : ClassTag]
       }
     case e:ProcessedDMEvent =>
       onProcessedDMEvent(e)
+    case e:MoveDeletedMessageMarkEvent => {
+      handleMoveDeletedMessageMarkEvent(e)
+    }
     case e:NewDMGeneratingVersionEvent =>
-      recoveredEventsCount_sinceLast_dmGeneratingVersion = 0 // reset counter
+      internalDurableState.recoveredEventsCount_sinceLast_dmGeneratingVersion = 0 // reset counter
       onNewDMGeneratingVersionEvent(e)
     case c:RecoveryCompleted =>
       onRecoveryCompleted()
     case event:E =>
       // Increment counter of events received since last NewDMGeneratingVersionEvent
-      recoveredEventsCount_sinceLast_dmGeneratingVersion = recoveredEventsCount_sinceLast_dmGeneratingVersion + 1
+      internalDurableState.recoveredEventsCount_sinceLast_dmGeneratingVersion = internalDurableState.recoveredEventsCount_sinceLast_dmGeneratingVersion + 1
       onReceiveRecover(event)
+    case offer: SnapshotOffer => {
+      if(!offer.snapshot.isInstanceOf[FullSnapshotState]) {
+        throw new RuntimeException(s"Snapshot is not of the expected type ${FullSnapshotState.getClass.getName}) but was ${offer.getClass.getName}" )
+      }
+      val offerState = offer.snapshot.asInstanceOf[FullSnapshotState]

```

Comment:
This code works, but is better written in Scala something like this:
offer.snaphot match {
  case offerState: FullSnapshotState => 
        onSnapshotOffer(SnapshotOffer(offer.metadata, offerState.userData))
  case _ => throw new RuntimeException(s"Snapshot is not...")
}